fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,290 @@
+        :param class_name:  interested class name in the given module objects list
+        :return:  tuple of name and class object of the given class name if class name found in the given module
+        objects, else returns an empty tuple.
+        """
+        for grpc_module in grpc_module_list:
+            for name, obj in inspect.getmembers(grpc_module, inspect.isclass):
+                logging.debug("Class Name : {} Object : {}".format(name, obj))
+                if name.lower().find(class_name.lower()) != -1:
+                    return name, obj
+        else:
+            return ()
+
+    def _get_method_from_stub_object(self, stub_object, comm_type, stub_name=''):
+        """
+        component that returns stub method from the given stub class objects, evaluates based on the given stub name and
+        communication type param.
+        :param stub_object:  stub Class object
+        :param comm_type:   communication type (i.e unary, server_stream,client_stream,bidirectional_stream)
+        :param stub_name:   name of the stub method in the stub class
+        :return: returns tuple of name and object of the stub method on success else returns None object
+        """
+        _found_stub_name = None
+        _found_stub_object = None
+        stub_elements = inspect.getmembers(stub_object)
+        for name, obj in stub_elements:
+            if name not in ['__class__', '__delattr__', '__dict__', '__doc__', '__eq__', '__format__', '__ge__',
+                            '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__',
+                            '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__',
+                            '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '__dir__']:
+
+                if str(obj).lower().find(comm_type) != -1 and name.lower().strip() == stub_name.lower().strip():
+                    logging.debug("found Stub name : {} and object : {}".format(name, obj))
+                    _found_stub_name = name
+                    _found_stub_object = obj
+                else:
+                    pass
+            else:
+                pass
+        return _found_stub_name, _found_stub_object
+
+    def _gRPC_Connect_Server(self, auth_type=None, channel=None, stub_class=None, certificate_path=None):
+        """
+
+        :param auth_type: Input Secure Connection type
+        :param channel: gRPC Channel
+        :param stub_class: a Stub Class object from imported pb2 library
+        :param certificate_path: path of the ssl/tls certificate
+        :return: stub class object
+        """
+        # Checking supplied Input Arguments
+        assert channel is not None, "channel argument should not be None type"
+        assert stub_class is not None, "stub class object should not be None type"
+        log = logging.getLogger(__name__)
```

Comment:
We can remove this if we add this line to the top

Reply:
ok